Automattic/woocommerce-payments
# Use original charge on payment details screen

Fixes #285 

#### Changes proposed in this Pull Request

* Redirect transaction list info button to payment details for original charge object
* Add get single charge endpoint to self-hosted sites
* Add fresh-data API spec to fetch charges from the React client
* Rename TransactionDetails to PaymentDetails to reduce ambiguity
* Use charge object when rendering payment details screen

#### Testing instructions

1. Sandbox the WP.com public API with 110-gh-woocommerce-payments-server
2. Under the transactions list, you should see the info button for transactions that are related to a charge
3. Click on the info button, you should be redirected to the payment details page for that charge 
    - It should be loaded from the network the first time you navigate to its details and cached on subsequent accesses
4. You should be able to navigate to a payment details page by manually inputting a valid change in the URL

_This page should look the same as the latest version of master_

![image](https://user-images.githubusercontent.com/7714042/70349923-78197b00-1844-11ea-9ee3-70af57254fbd.png)
![image](https://user-images.githubusercontent.com/7714042/70349933-7bad0200-1844-11ea-97d8-9fcc5055a4c0.png)
![image](https://user-images.githubusercontent.com/7714042/70349937-7d76c580-1844-11ea-8450-a3870893322b.png)


-------------------
